test/recipes/20-test_pkeyutl.t

```diff
@@ -15,6 +15,10 @@ use OpenSSL::Test::Utils;
 
 setup("test_pkeyutl");
 
+plan skip_all => "test_pkeyutl needs EC, SM2 and SM3 enabled"
+    if disabled("ec") || disabled("sm2") || disabled("sm3");
```

Comment:
Considering the name of this test, This seems wrong, even though the *current* contents are only about ec / sm2 / sm3.  I would much prefer that the relevant section (even it's everything) gets wrapped in this sort of construct:

``` perl
 SKIP:
    {
        skip "Skipping tests that require EC, SM2 and SM3", {n}
            if disabled("ec") || disabled("sm2") || disabled("sm3");

        # ... test code

    }
```

I think that sort of construct is more encouraging toward those who want to add more tests for other algorithms, and also becomes a template for how to do so.

Reply:
Ok - done.

Reply:
(at some point, the two functions there will have to be rewritten to be able to handle other algos as well, but no matter, that's for another PR)